packages/client/tests/functional/_utils/getTestSuiteInfo.ts

```diff
@@ -0,0 +1,137 @@
+import fs from 'fs-extra'
+import path from 'path'
+
+import { keys } from '../../../../../helpers/blaze/keys'
+import { map } from '../../../../../helpers/blaze/map'
+import { matrix } from '../../../../../helpers/blaze/matrix'
+import { merge } from '../../../../../helpers/blaze/merge'
+import type { TestSuiteMeta } from './setupTestSuiteMatrix'
+
+export type TestSuiteMatrix = { [K in string]: string }[][]
+export type TestSuiteConfig = ReturnType<typeof getTestSuiteConfigs>[number]
+
+/**
+ * Get the generated test suite name, used for the folder name.
+ * @param suiteMeta
+ * @param suiteConfig
+ * @returns
+ */
+export function getTestSuiteFullName(suiteMeta: TestSuiteMeta, suiteConfig: TestSuiteConfig) {
+  let name = `${suiteMeta.suiteName} - ${suiteConfig['#PROVIDER']}`
+
+  name += ` - [`
+  if (suiteConfig['#FEATURES']) {
+    name += `${suiteConfig['#FEATURES']}`
+  }
+
+  if (suiteConfig['#EXTRA_FEATURES']) {
+    name += `${suiteConfig['#EXTRA_FEATURES']}`
+  }
+  name += `]`
+
+  return name
+}
+
+/**
+ * Get the generated test suite features, used for client generation.
+ * @param suiteConfig
+ * @returns
+ */
+export function getTestSuitePreviewFeatures(suiteConfig: TestSuiteConfig) {
+  // eslint-disable-next-line prettier/prettier
+  return [...(suiteConfig['#FEATURES']?.split(', ') ?? []), ...(suiteConfig['#EXTRA_FEATURES']?.split(', ') ?? [])]
```

Comment:
**(non-blocking)** I think this line needs rethinking, its too complicated.

Reply:
It's now on two lines so it should be easier to digest.